### DISCUSSION TITLE
Update item and create relations if exists, otherwise skip

### DISCUSSION THREAD

**Discussion by reteps** [prisma/prisma#18684] [Q&A] [answered]
Hi, I want to execute a transaction like this:

```ts
const updates = newUsers.map(newUser => {
  const newPosts = // [...]
  return prisma.users.update({
              where: {
                 name: newUser.name
              },
              data: {
                status: newUser.status,
                posts: {
                  create: newPosts,
                },
              }
  });
})
 ```
 
 If a user exists in the database, I want to:

-  Update the field `status` on the user. Create a list of associated posts for that user.

If the user `newUser` does not exists in the database, I want to:
- **Do Nothing**

Currently, when I do:

```ts
await ctx.prisma.$transaction(updates);
```

It throws an error saying that it cannot create the posts for a (user that doesn't exist), since a previous operation (e.g. the update) failed. However, if the update returns nothing, I don't want it to create any posts at all. How can I do this? 

ChatGPT has suggested three workarounds:

- Do all lookups for usernames in a single transaction, then all updates in a single transaction based on the returned users
- Use `Promise.all` instead of a transaction
- Use raw SQL

Is there a way that I don't need to do this?

I am imagining an upsert with an empty `create` clause.

**Comment by nurul3101** [comment, marked as answer]
Hello @reteps 👋 

At the moment, an upsert with an empty create clause would throw an error.

The most straightforward solution, I think would be to do a check beforehand to validate if a user exists (With a [find unique](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique) query) and then, depending on the result, either invoke an update query or do nothing in case the record doesn't exist.

**Reply by nurul3101** [reply]
Additionally, would [updateMany](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#updatemany) work for your usecase?

**Reply by baudehlo** [reply]
I thumbs down'd this, because it's a clear race condition.

The only way to do it safely is with updateMany.
